src/main/java/software/amazon/event/kafkaconnector/EventBridgeWriter.java

```diff
@@ -180,75 +205,58 @@ public void shutDownEventBridgeClient() {
    * @param record Record to be sent to EventBridge
    * @return the PutEventsRequestEntry ready to send
    */
-  private Optional<PutEventsRequestEntry> createPutEventsEntry(EventBridgeWriterRecord record) {
+  private Either<EventBridgeWriterErrorSinkRecord, Pair<PutEventsRequestEntry, SinkRecord>>
```

Comment:
Hi @agebhar1, I think the usage of Either with a Pair makes the code quite verbose and one needs to dive deep to understand the flow. My initial thought was relating to use/create something similar to Kotlin's [Result](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-result/). In that way we can return a List<Result> and filter/partition on .isSuccess / .isFailure respectively. Wdyt?

Reply:
We can then also reuse this Result class for putItems itself. Rather than providing just a list of errors we provide Result with Success and Errors and the Tasks filters for the errors.

Reply:
Already did this but removed it after saw the `Either` type of the AWS SDK. But fully agree with a Kotlin like `Result` type.